src/metric/event.rs

```diff
@@ -41,6 +43,8 @@ pub enum Event {
         encoding: Encoding,
         /// Encoded payload.
         bytes: Vec<u8>,
+        /// Header values used by some sinks to send metadata
+        headers: HashMap<Vec<u8>, Vec<u8>>,
```

Comment:
Could make `Option<HashMap<Vec<u8>, Vec<u8>>>`, probably worth it to avoid allocations.

Reply:
I am 9/10s sure @blt would call you out on the allocations bit. :)